$NetBSD: patch-meson.build,v 1.24 2025/05/27 19:07:43 adam Exp $

- On Darwin, optionally use Cocoa (needed for X11 users).

- Don't insist that Python binary is called "python3".
  (Requires meson >= 0.50.0)

- Don't install bash completion scripts.

- Tolerate getxattr not existing (smartOS)
  https://gitlab.gnome.org/GNOME/glib/merge_requests/888

- Check also for sys/ptrace.h header file.

--- meson.build.orig	2025-05-20 16:22:25.000000000 +0000
+++ meson.build
@@ -438,6 +438,7 @@ headers = [
   'sys/mount.h',
   'sys/param.h',
   'sys/prctl.h',
+  'sys/ptrace.h',
   'sys/resource.h',
   'sys/select.h',
   'sys/statfs.h',
@@ -1010,7 +1011,7 @@ if host_system == 'darwin'
                                       #error "Detected GNUstep, not Cocoa"
                                       #endif''',
                                    name : 'Mac OS X Cocoa support')
-
+  glib_have_cocoa = glib_have_cocoa and get_option('use_cocoa')
   if glib_have_cocoa
     glib_conf.set('HAVE_COCOA', true)
     osx_ldflags += ['-Wl,-framework,Foundation', '-Wl,-framework,AppKit']
@@ -2379,18 +2380,17 @@ if host_system != 'windows' and get_opti
 
   # try glibc
   if cc.has_function('getxattr') and cc.has_header('sys/xattr.h')
+    glib_conf.set('HAVE_XATTR', 1)
     glib_conf.set('HAVE_SYS_XATTR_H', 1)
     glib_conf_prefix = glib_conf_prefix + '#define @0@ 1\n'.format('HAVE_SYS_XATTR_H')
   #failure. try libattr
   elif cc.has_header_symbol('attr/xattr.h', 'getxattr')
+    glib_conf.set('HAVE_XATTR', 1)
     glib_conf.set('HAVE_ATTR_XATTR_H', 1)
     glib_conf_prefix = glib_conf_prefix + '#define @0@ 1\n'.format('HAVE_ATTR_XATTR_H')
     xattr_dep = [cc.find_library('xattr')]
-  else
-    error('No getxattr implementation found in C library or libxattr')
   endif
 
-  glib_conf.set('HAVE_XATTR', 1)
   if cc.compiles(glib_conf_prefix + '''
                  #include <stdio.h>
                  #ifdef HAVE_SYS_TYPES_H
@@ -2491,7 +2491,7 @@ endif
 
 glib_conf.set('HAVE_PROC_SELF_CMDLINE', have_proc_self_cmdline)
 
-python = import('python').find_installation()
+python = import('python').find_installation('@PYTHONBIN@')
 
 # Sets the shebang on script files. This has been changed many times including
 # using env, the absolute path to the Python executable, and using both python
@@ -2510,7 +2510,7 @@ python_test_libraries_built = meson.proj
 
 # Determine which user environment-dependent files that we want to install
 bash = find_program('bash', required : false)
-have_bash = bash.found() # For completion scripts
+have_bash = false
 bash_comp_dep = dependency('bash-completion', version: '>=2.0', required: false)
 have_sh = find_program('sh', required : false).found() # For glib-gettextize
 have_pkg_config = find_program('pkg-config', required: false).found()
